roles/keepalived/tasks/main.yml

```diff
@@ -42,11 +43,22 @@
   notify: "restart keepalived"
   tags: keepalived_conf, keepalived
 
+- name: Parse keepalived_instances as YAML
+  ansible.builtin.set_fact:
+    keepalived_instances_parsed: "{{ keepalived_instances | from_yaml }}"
```

Comment:
this task seems superfluous, why are you using the `from_yaml` filter here?

Reply:
I'm using a Jinja2 template to create the keepalived_instances variable in my group_vars. Indeed, this filter should be done in group_vars, not here.